From 9a60302d474a1c482a769a469bda22361fd73e61 Mon Sep 17 00:00:00 2001
From: Luo Jie <luoj@qti.qualcomm.com>
Date: Mon, 13 Jul 2026 11:15:58 +0800
Subject: [PATCH] driver: clk: ipq-cmn-pll: Mark gated output clocks as
 CLK_IGNORE_UNUSED

Mark these clocks with CLK_IGNORE_UNUSED so the framework leaves them
alone during the unused-clock cleanup pass.

Change-Id: Iaec7d952ec02440c769ae08e056704f848858ba0
Signed-off-by: Luo Jie <luoj@qti.qualcomm.com>
---
 drivers/clk/qcom/ipq-cmn-pll.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/clk/qcom/ipq-cmn-pll.c b/drivers/clk/qcom/ipq-cmn-pll.c
index dc1fd3f78c04..e3680fa91125 100644
--- a/drivers/clk/qcom/ipq-cmn-pll.c
+++ b/drivers/clk/qcom/ipq-cmn-pll.c
@@ -1302,7 +1302,7 @@ static struct clk_hw *ipq_cmn_pll_pon_refclk_register(struct platform_device *pd
 	init.parent_data = &pdata;
 	init.num_parents = 1;
 	init.ops = &clk_pon_refclk_ops;
-	init.flags = 0;
+	init.flags = CLK_IGNORE_UNUSED;
 
 	pon_clk->hw.init = &init;
 	pon_clk->regmap = regmap;
@@ -1407,7 +1407,7 @@ static struct clk_hw *ipq_cmn_pll_pcs_register(struct platform_device *pdev,
 	init.parent_data = &pdata;
 	init.num_parents = 1;
 	init.ops = &clk_pcs_clk_ops;
-	init.flags = 0;
+	init.flags = CLK_IGNORE_UNUSED;
 
 	pcs_clk->hw.init = &init;
 	pcs_clk->regmap = regmap;
@@ -1477,7 +1477,7 @@ static struct clk_hw *ipq_cmn_pll_eth_pon_register(struct platform_device *pdev,
 	init.parent_data = &pdata;
 	init.num_parents = 1;
 	init.ops = &clk_eth_pon_ops;
-	init.flags = 0;
+	init.flags = CLK_IGNORE_UNUSED;
 
 	eth_pon_clk->hw.init = &init;
 	eth_pon_clk->regmap = regmap;
@@ -1541,7 +1541,7 @@ static struct clk_hw *ipq_cmn_pll_register_fixed_gate(struct device *dev,
 						    NULL, NULL,
 						    &fixed->hw, &clk_fixed_rate_ops,
 						    &gate->hw, &clk_gate_ops,
-						    0);
+						    CLK_IGNORE_UNUSED);
 }
 
 static int ipq_cmn_pll_register_clks(struct platform_device *pdev)
-- 
2.34.1

